src/NuGet.Clients/NuGet.CommandLine/Common/NuGetTestMode.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace NuGet.Common
+{
+    // Note: we may delete this logic once we no longer depend on NuGet.Core's PackageServer type,
+    // as it is embedded in NuGet.Protocol.Core.Types.UserAgent.
+    internal static class NuGetTestMode
+    {
+        private const string _testModeEnvironmentVariableName = "NuGetTestModeEnabled";
```

Comment:
The code is not setting the env var anywhere, only reading it if available.
This line is setting it within the scope of the script only: https://github.com/NuGet/NuGet.Client/pull/135/files#diff-3abd5921200b239022d9c6e0240a9692R35